src/br/com/rtech/service/CreateCourseService.java

```diff
@@ -0,0 +1,51 @@
+package br.com.rtech.service;
+
+import br.com.rtech.model.Course;
+import br.com.rtech.model.SubCategory;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class CreateCourseService {
+    public static List<Course> generateObjectCourse(String path, List<SubCategory> subCategories) {
+        List<Course> courses = new ArrayList<>();
+        try(Scanner scanner = new Scanner(new File(path))) {
+            // skip the labels
+            scanner.nextLine();
+            while (scanner.hasNext()) {
+                String[] atributesCourse = scanner.nextLine().split(",");
+                String name = atributesCourse[0];
+                String code = atributesCourse[1];
+                int estimatedTimeCourseCompletion = Integer.parseInt(atributesCourse[2]);
+                String visibility = atributesCourse[3];
+                String targetPublic = atributesCourse[4];
+                String instructorsName = atributesCourse[5];
+                String syllabus = atributesCourse[6];
+                String skills = atributesCourse[7];
+                String subcategoryCode = atributesCourse[8];
+                for (SubCategory subCategory: subCategories) {
```

Comment:
Qual estrutura de dados você poderia usar aqui para evitar que fosse feito um for em subcategorias para cada uma das linhas do arquivo?

Reply:
Refleti e ainda não cheguei em uma conclusão.
Estou utilizando duas estruturas de repetição. A primeira é para a leitura das linhas do CSV e criar os atributos, já a segunda é para pesquisar se o código existe na lista de subcategorias. Como a lista de subcategorias na muda ou seja é a mesma em todas iterações do for externo, a ideia seria que eu iterasse somente uma vez e armazenasse em uma estrutura de dados, e a partir disso, eu somente fizesse uma pesquisa pelo código?